### PR TITLE
Mip dna v8.2

### PR DESCRIPTION
**Pull request by henrikstranneheim**
This PR makes sure that cg is MIP 8 compatible, adds new to files store in housekeeper and updates what to load in Scout.

**CLI Changes**
`family -> case` for `workflow <workflow_name> deliver inbox`

**How to prepare for test**:
- [x] install on stage: `bash /home/proj/stage/servers/resources/hasta.scilifelab.se/update-cg-stage.sh mip-dna-v8.2`
- [x] install on stage: `bash /home/proj/stage/servers/resources/hasta.scilifelab.se/update-trailblazer-stage.sh mip-dna-v8.2`
- [x] install on stage: `bash /home/proj/stage/servers/resources/hasta.scilifelab.se/update-servers-stage.sh mip-dna-v8.2`
- [x] install on stage: `bash /home/proj/stage/servers/resources/hasta.scilifelab.se/update-housekeeper-stage.sh feat/add-new-tag`
- [x] install on stage: `bash /home/proj/stage/servers/resources/hasta.scilifelab.se/update-scout-cli-stage.sh master`
- [x] activate stage: `us`
- [x] Add new tags to housekeeper: 
  - `housekeeper add tag multiqc-html`
  - `housekeeper add tag multiqc-json`
  - `housekeeper add tag pedigree-yaml`
  - `housekeeper add tag exe-ver`
  - `housekeeper add tag cram`
  - `housekeeper add tag cram-index`
  - `housekeeper add tag chromograph`

**How to test**:
*vitalmouse - wgs, trio*
1. Run following command: `cg workflow mip-dna -p low -c vitalmouse`
1.  Run following command: `cg workflow mip-dna store analysis /home/proj/stage/rare-disease/cases/vitalmouse/analysis/vitalmouse_config.yaml`
1. Run following command: `housekeeper get -V vitalmouse`
1. Run the following command: `cg upload scout vitalmouse`
1. Run the following command: `cg workflow mip-dna deliver inbox vitalmouse`

**Expected test outcome**:
1. `cg` should start a MIP 8.2.0 analysis run and complete without errors
1. `cg` should store a MIP 8.2.0 analysis files, including new files:
  - pedigree_yaml
  - multiqc_html
  - multiqc_json
  - version_collect
  - chromograph
1. Make sure that all new files and the IGV files for samples now is stored in the CRAM format with ".cram" and ".crai" file endings
1. The case should be uploaded to Scout
1. Cram files and index should be linked to inbox

*onemite - wgs, single*
1. Run following command: `cg workflow mip-dna -p low -c onemite`
1.  Run following command: `cg workflow mip-dna store analysis /home/proj/stage/rare-disease/cases/onemite/analysis/onemite_config.yaml`
1. Run following command: `housekeeper get -V onemite`
1. Run the following command: `cg upload scout onemite`
1. Run the following command: `cg workflow mip-dna deliver inbox onemite`

**Expected test outcome**:
1. `cg` should start a MIP 8.2.0 analysis run and complete without errors
1. `cg` should store a MIP 8.2.0 analysis files, including new files:
  - pedigree_yaml
  - multiqc_html
  - multiqc_json
  - version_collect
1. Make sure that all new files and the IGV files for samples now is stored in the CRAM format with ".cram" and ".crai" file endings
1. The case should be uploaded to Scout
1. Cram files and index should be linked to inbox

*strongbison - wes, trio*
1. Run following command: `cg workflow mip-dna -p low -c strongbison`
1.  Run following command: `cg workflow mip-dna store analysis /home/proj/stage/rare-disease/cases/strongbison/analysis/strongbison_config.yaml`
1. Run following command: `housekeeper get -V strongbison`
1. Run the following command: `cg upload scout  strongbison`
1. Run the following command: `cg workflow mip-dna deliver inbox strongbison`

**Expected test outcome**:
1. `cg` should start a MIP 8.2.0 analysis run and complete without errors
1. `cg` should store a MIP 8.2.0 analysis files, including new files:
  - pedigree_yaml
  - multiqc_html
  - multiqc_json
  - version_collect
1. Make sure that all new files and the IGV files for samples now is stored in the CRAM format with ".cram" and ".crai" file endings
1. The case should be uploaded to Scout
1. Cram files and index should be linked to inbox

*gladthrush - wes, single*
1. Run following command: `cg workflow mip-dna -p low -c gladthrush`
1.  Run following command: `cg workflow mip-dna store analysis /home/proj/stage/rare-disease/cases/gladthrush/analysis/gladthrush_config.yaml`
1. Run following command: `housekeeper get -V gladthrush`
1. Run the following command: `cg upload scout gladthrush`
1. Run the following command: `cg workflow mip-dna deliver inbox gladthrush`

**Expected test outcome**:
1. `cg` should start a MIP 8.2.0 analysis run and complete without errors
1. `cg` should store a MIP 8.2.0 analysis files, including new files:
  - pedigree_yaml
  - multiqc_html
  - multiqc_json
  - version_collect
1. Make sure that all new files and the IGV files for samples now is stored in the CRAM format with ".cram" and ".crai" file endings
1. The case should be uploaded to Scout
1. Cram files and index should be linked to inbox

**Review:**
- [x] code approved by @patrikgrenfeldt 
- [x] tests executed by @henrikstranneheim 
- [x] update AM 1060 @henrikstranneheim 
- [x] updated AM 1203 @henrikstranneheim 
- [x] validation plan written
- [x] validation plan executed
- [x] relevant staff informed
- [x] "Merge and deploy" approved by @patrikgrenfeldt 
Thanks for filling in who performed the code review and the test!

**Implementation plan**
- [x] merge and deploy [servers](https://github.com/Clinical-Genomics/servers/pull/279) @patrikgrenfeldt 
- [x] merge and deploy [trailblazer](https://github.com/Clinical-Genomics/trailblazer/pull/74) @patrikgrenfeldt 
- [x] merge and deploy [housekeeper ](https://github.com/Clinical-Genomics/housekeeper/pull/55) @patrikgrenfeldt 
- [x] merge and deploy [scout](https://github.com/Clinical-Genomics/scout/pull/1722) @northwestwitch 
- [x] install MIP 8.2 in P_mip8.2_rd-dna and S_mip8.2_rd-dna
- [ ] email bioinfoit
- [ ] Merge and deploy PR 

This is major **version bump** because we are changing the cg CLI in an non backwards compatible manner.

### CODE REVIEW

**Comment by henrikstranneheim** [comment]
**How to test:**
vitalmouse - wgs, trio
1. Run following command: cg workflow mip-dna -p low -c vitalmouse
1.  Run following command: `cg workflow mip-dna store analysis /home/proj/stage/rare-disease/cases/vitalmouse/analysis/vitalmouse_config.yaml`
1. Run following command: `housekeeper get -V vitalmouse`
1. Run the following command: `cg upload scout vitalmouse`
1. Run the following command: `cg workflow mip-dna deliver inbox vitalmouse`


**Expected test outcome**:
1. `cg` should start a MIP 8.2.0 analysis run and complete without errors

Screenshot 2020-02-06 at 14 04 43


1. `cg` should store a MIP 8.2.0 analysis files, including new files: 
  - pedigree_yaml
  - multiqc_html
  - multiqc_json
  - version_collect
  - chromograph



1. Make sure that all new files and the IGV files for samples now is stored in the CRAM format with ".cram" and ".crai" file endings.


1. The case should be uploaded to Scout. 



1. Cram files and index should be linked to inbox.

*onemite - wgs, single*
1. Run following command: `cg workflow mip-dna -p low -c onemite`

**Expected test outcome**:
1. `cg` should start a MIP 8.2.0 analysis run and complete without errors



*strongbison - wes, trio*
1. Run following command: `cg workflow mip-dna -p low -c strongbison`

**Expected test outcome**:
1. `cg` should start a MIP 8.2.0 analysis run and complete without errors 



NOTE: Analysisrunstatus fails due to that the CSQ flag "FLAGS" does not have a value for the SV select file. This is related to the data for this trio and not a systematic problem in production and can therefore be safely ignored. 

*gladthrush - wes, single*
1. Run following command: `cg workflow mip-dna -p low -c gladthrush`

**Expected test outcome**:
1. `cg` should start a MIP 8.2.0 analysis run and complete without errors



NOTE: Fails on expected coverage but this expected and related to the actual sequencing depth for this sample and not related to this PR

**Comment by henrikstranneheim** [comment]
@patrikgrenfeldt All tests pass even the one for vitalmouse upload to Scout I am just waiting for a fix for the alignment_path so that I can see the CRAM files in IGV in Scout, but the cg code seems to work. It would be great if you could allocate some time for a code review as soon as possible.

**Comment by patrikgrenfeldt** [comment]
Do the automatic tests pass when run together with correct version of HK and TB?
I suggest we add a test case for running pytest manually with the correct versions since the Travis checks fail and maybe are expected to fail. What do you say @henrikstranneheim

**Comment by henrikstranneheim** [comment]
Yes, they do locally. Sounds proper, do you want to set it up in the travis file or what did you have in mind? I guess I could show the test suite of cg passes on hasta when the proper TB, HK is installed.

File: cg/apps/tb/add.py [hunk 1]
```diff
@@ -39,145 +40,196 @@ def add_analysis(cls, config_stream):
...
+                "archive": False,
+            },
+            {
+                "path": sampleinfo_data["peddy"]["sex_check"],
+                "tags": ["peddy", "sex-check"],
+                "archive": False,
+            },
+            {
+                "path": sampleinfo_data["version_collect"],
+                "tags": ["executables-version"],
```
Comment (patrikgrenfeldt) [inline cg/apps/tb/add.py]: This is a really long tag..

File: cg/apps/tb/add.py [hunk 2]
```diff
@@ -39,145 +40,196 @@ def add_analysis(cls, config_stream):
...
-                    'archive': False,
-                })
-
-            cytosure_path = sample_data['vcf2cytosure']
-            data.append({
-                'path': cytosure_path,
-                'tags': ['vcf2cytosure', sample_data['id']],
-                'archive': False,
-            })
+                    mt_bai_path = mt_bam_path.replace(".bam", ".bai")
```
Comment (patrikgrenfeldt) [inline cg/apps/tb/add.py]: Bam?

File: tests/meta/conftest.py [hunk 3]
```diff
@@ -136,47 +137,160 @@ def analysis_store(base_store, analysis_family):
...
+class MockLims:
+    """Mock lims fixture"""
+
+    lims = None
+
+    def __init__(self):
+        self.lims = self
+
 
 class MockDeliver:
```
Comment (patrikgrenfeldt) [inline tests/meta/conftest.py]:
```suggestion
class MockDeliver(DeliverAPI):
```
To get the rest of the API "for free"

File: tests/meta/deliver/test_delivery_api.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    deliver_tags = ["case-tag"]
+
+    # WHEN we call get_post_analysis_files with matching case and file-tags and version is supplied
+    version = "a_date"
+
+    bundle_version_files = deliver_api.get_post_analysis_files(
+        case, version, deliver_tags
+    )
+
+    ## THEN housekeeper files for those tags with the version should be returned
```
Comment (patrikgrenfeldt) [inline tests/meta/deliver/test_delivery_api.py]:
```suggestion
    # THEN housekeeper files for those tags with the version should be returned
```

File: tests/meta/deliver/test_delivery_api.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+    # WHEN we call get_post_analysis_case_files with matching case and file-tags and version
+    version = None
+    case_files = deliver_api.get_post_analysis_case_files(case, version, deliver_tags)
+
+    # THEN we should get those matching files back
+    assert case_files
+
+
+def test_get_post_analysis_case_files_only_sample_tags(analysis_family, deliver_api):
```
Comment (patrikgrenfeldt) [inline tests/meta/deliver/test_delivery_api.py]: This test and the previous test has the same GIVEN-WHEN-THEN. The normal suggestion in such a case is to remove one of the tests. I think we could update the GIVEN-WHEN-THENs so it clear what tests are testing ?

File: tests/meta/upload/test_meta_upload_scoutapi.py [hunk 6]
```diff
@@ -8,6 +8,17 @@
 from cg.meta.upload.scoutapi import UploadScoutAPI
 from cg.store import Store
 
+CASE_FILE_PATHS = ["multiqc"]
+
+RESULT_KEYS = [
+    "family",
```
Comment (patrikgrenfeldt) [inline tests/meta/upload/test_meta_upload_scoutapi.py]: Family?

**Comment by patrikgrenfeldt** [comment]
I can't find any PR for housekeeper/feat/add-new-tag, I guess we need one?

File: cg/apps/tb/add.py (hunk 1 above)
Comment (henrikstranneheim) [reply cg/apps/tb/add.py]: Shortened it to exe-ver

File: cg/apps/tb/add.py (hunk 2 above)
Comment (henrikstranneheim) [reply cg/apps/tb/add.py]: yes, that file is downsampled to 150 x and only cover the 16 kb of the MT so it is rather small. No need to convert it in the workflow to CRAM

File: tests/meta/conftest.py (hunk 3 above)
Comment (henrikstranneheim) [reply tests/meta/conftest.py]: Nice!

File: tests/meta/deliver/test_delivery_api.py (hunk 5 above)
Comment (henrikstranneheim) [reply tests/meta/deliver/test_delivery_api.py]: Added a more clear comment

File: tests/meta/upload/test_meta_upload_scoutapi.py (hunk 6 above)
Comment (henrikstranneheim) [reply tests/meta/upload/test_meta_upload_scoutapi.py]: Yes, I think the mock analysis is using the family key and I do not want to change it in this PR as it is used by many tests.

Comment (patrikgrenfeldt) [reply tests/meta/upload/test_meta_upload_scoutapi.py]: seems like a scout thing, might be beyond scope of this PR

**Comment by patrikgrenfeldt** [comment]
deployed on hasta and clinical-db

**Pull request merged**